src/sqlite3gen.cpp

```diff
@@ -1110,172 +1872,201 @@ static void generateSqlite3ForMember(const MemberDef *md, const Definition *def)
+
   for (mli.toFirst();(md=mli.current());++mli)
   {
+    // TODO: necessary? just tracking what xmlgen does; xmlgen says:
     // namespace members are also inserted in the file scope, but
     // to prevent this duplication in the XML output, we filter those here.
     if (d->definitionType()!=Definition::TypeFile || md->getNamespaceDef()==0)
     {
-      count++;
+      generateSqlite3ForMember(md, scope_refid, d);
     }
   }
-  if (count==0) return; // empty list
-  for (mli.toFirst();(md=mli.current());++mli)
+}
+
+static void associateAllClassMembers(const ClassDef *cd, struct Refid scope_refid)
+{
+  if (cd->memberNameInfoSDict())
   {
-    // namespace members are also inserted in the file scope, but
-    // to prevent this duplication in the XML output, we filter those here.
-    //if (d->definitionType()!=Definition::TypeFile || md->getNamespaceDef()==0)
+    MemberNameInfoSDict::Iterator mnii(*cd->memberNameInfoSDict());
+    MemberNameInfo *mni;
+    for (mnii.toFirst();(mni=mnii.current());++mnii)
     {
-      generateSqlite3ForMember(md,d);
+      MemberNameInfoIterator mii(*mni);
+      MemberInfo *mi;
+      for (mii.toFirst();(mi=mii.current());++mii)
+      {
+        MemberDef *md = mi->memberDef;
+        QCString qrefid = md->getOutputFileBase() + "_1" + md->anchor();
+        associateMember(md, insertRefid(qrefid), scope_refid);
+      }
     }
   }
 }
 
-
+// many kinds: category class enum exception interface
+// module protocol service singleton struct type union
+// enum is Java only (and is distinct from enum memberdefs)
 static void generateSqlite3ForClass(const ClassDef *cd)
 {
+  // NOTE: Skeptical about XML's version of these
+  // 'x' marks missing items XML claims to include
+
+  // + brief description
+  // + detailed description
+  // + template argument list(s)
+  // + include file
+  // + member groups
+  // x inheritance DOT diagram
   // + list of direct super classes
   // + list of direct sub classes
-  // + include file
   // + list of inner classes
-  // - template argument list(s)
-  // + member groups
+  // x collaboration DOT diagram
   // + list of all members
-  // - brief description
-  // - detailed description
-  // - inheritance DOT diagram
-  // - collaboration DOT diagram
-  // - user defined member sections
-  // - standard member sections
-  // - detailed member documentation
+  // x user defined member sections
+  // x standard member sections
+  // x detailed member documentation
   // - examples using the class
 
   if (cd->isReference())        return; // skip external references.
   if (cd->isHidden())           return; // skip hidden classes.
   if (cd->name().find('@')!=-1) return; // skip anonymous compounds.
   if (cd->templateMaster()!=0)  return; // skip generated template instances.
 
-  msg("Generating Sqlite3 output for class %s\n",cd->name().data());
+  struct Refid refid = insertRefid(cd->getOutputFileBase());
+
+  // can omit a class that already has a refid
```

Comment:
Unless there are conditions under which we may encounter the class refid before parsing the class?

Also, this same basic issue (but for pages) recurs on line 2470.